packages/react-components/react-utilities/src/utils/types.ts

```diff
@@ -31,3 +31,23 @@ export type UnionToIntersection<U> = (U extends unknown ? (x: U) => U : never) e
  * If type T includes `null`, remove it and add `undefined` instead.
  */
 export type ReplaceNullWithUndefined<T> = T extends null ? Exclude<T, null> | undefined : T;
+
+/**
+ * This type should be used in place of `React.RefAttributes<T>` in all components that specify `ref` prop.
+ *
+ * If user is using React 18 types v >=18.2.61, they will run into type issues of incompatible refs, using this type mitigates this issues across react type versions.
+ *
+ * @remarks
+ *
+ * React 18 types introduced Breaking Changes to the `RefAttributes` interface as patch release.
+ * These changes were release in @types/react@18.2.61 (replacing ref with `LegacyRef`)
+ *  - PR: https://github.com/DefinitelyTyped/DefinitelyTyped/pull/68720
+ *  - shipped definitions: https://app.unpkg.com/@types/react@18.2.61/files/index.d.ts
+ *
+ * In React 19 types this was "reverted" back to the original `Ref<T>` type.
+ * In order to maintain compatibility with React 17,18,19, we are forced to use our own version of `RefAttributes`.
+ *
+ */
+export interface RefAttributes<T> extends React.Attributes {
+  ref?: React.Ref<T> | undefined;
```

Comment:
this was the type in:
- `@types/react` v17
- `@types/react` v18
- `@types/react` v18.2.61 changed the type to `LegacyRef`
- `@types/react` v19 reverted this change and again uses `Ref`